runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/config/NamespaceConfig.java

```diff
@@ -52,5 +56,26 @@ public static NamespaceConfigBuilder<NamespaceConfig> builder()
         this.guards = requireNonNull(guards);
         this.vaults = requireNonNull(vaults);
         this.catalogs = requireNonNull(catalogs);
+        this.resources = resolveResources();
+    }
+
+    private List<String> resolveResources()
+    {
+        List<String> resources = new LinkedList<>();
+        for (CatalogConfig catalog : catalogs)
+        {
+            if (FILESYSTEM.equals(catalog.type))
+            {
+                resources.addAll(catalog.options.resources);
+            }
+        }
+        for (VaultConfig vault : vaults)
+        {
+            if (FILESYSTEM.equals(vault.type))
+            {
+                resources.addAll(vault.options.resources);
+            }
+        }
+        return resources;
```

Comment:
This approach assumes global knowledge of all possible implementations and assumes their behavior, such that a catalog or vault other than `filesystem` cannot contribute resources to be watched.

We need a more general strategy instead.

Please include all resources from any `binding`, `guard`, `vault`, `catalog` or `telemetry` `exporter`.

Note: instead of calling a non-static method from the constructor that relies on constructor field assignment, please make this method static and pass all required state as parameters, such as `bindings`, `guards`, `vaults`, `catalogs`, and `telemetry`.